### PR TITLE
Fix handling of DVD commands that read 0 bytes

### PR DESCRIPTION
**Pull request by JosJuice**
Fixes a regression from the chunking PR that was making a few games hang, and removes an error message that was showing up even though no error was happening.

### CODE REVIEW

**Comment by JMC47** [comment]
Fixes Zapper: One Wicked Cricket

**Comment by delroth** [comment]
LGTM

File: Source/Core/Core/HW/DVD/DVDInterface.cpp [hunk 1]
```diff
@@ -1222,6 +1222,8 @@ void ScheduleReads(u64 offset, u32 length, const DiscIO::Partition& partition, u
     // The last chunk may be short
     chunk_length = std::min(chunk_length, length);
 
+    // TODO: If the emulated software requests 0 bytes of data, should we seek or not?
```
Comment (shuffle2) [inline Source/Core/Core/HW/DVD/DVDInterface.cpp]: What was the old behavior? Did you run gc service disk against these changes?

Comment (JosJuice) [reply Source/Core/Core/HW/DVD/DVDInterface.cpp]: We were seeking before the chunking PR, and we seek with this PR. (With the chunking PR, we were essentially seeking except we were doing it too quickly, but it doesn't really matter since the game would hang anyway.)

I didn't run it before, but I ran the optical disc drive test now. It passes in master and in this PR, and it doesn't use any 0-length reads.

Comment (JMC47) [reply Source/Core/Core/HW/DVD/DVDInterface.cpp]: I tested a bunch of the other weird edge-cases in Dolphin too.  This is the only game where I ran into a zero byte read.

Comment (JosJuice) [reply Source/Core/Core/HW/DVD/DVDInterface.cpp]: IIRC, Gecko OS does a zero byte read when launching a game. I haven't noticed anything else doing it, though.
